### ISSUE TITLE
docker pull nvcr.io/nvidia/pytorch:19.04-py3 fails with "Authorization Required"

### ISSUE THREAD

**Issue by thepaulm** [NVIDIA/retinanet-examples#16]
Hello - as detailed over [here](https://github.com/NVIDIA/retinanet-examples/issues/15) apparently there is some layer in your docker container requiring extra authorization?

```
gpu:~$ docker pull nvcr.io/nvidia/pytorch:19.04-py3
19.04-py3: Pulling from nvidia/pytorch
34667c7e4631: Pull complete 
d18d76a881a4: Pull complete 
119c7358fbfc: Pull complete 
2aaf13f3eff0: Pull complete 
202fa0f8874b: Pull complete 
3b700a61ede6: Pull complete 
87e6ca450d3f: Pull complete 
a1e76dce1aec: Pull complete 
9b91fa2f9276: Pull complete 
b5877a9add73: Pull complete 
bab74df105f1: Pull complete 
534bbf505504: Pull complete 
4956bf3bbbb9: Pull complete 
f4371944c97d: Pull complete 
4615a735431d: Pull complete 
5db2639932b5: Pull complete 
629d5c9d75a4: Pull complete 
8071b94b5429: Pull complete 
6eb8eba2ad5a: Pull complete 
e32e86c15b8b: Pull complete 
08db5b51b243: Pull complete 
f71ce95fb406: Pull complete 
3498ed8c5685: Pull complete 
62819d8896c1: Pull complete 
34bc85bf8bef: Pull complete 
dcdd5f85017f: Pull complete 
1e228f756b55: Downloading [=======================================>           ]  403.1MB/513MB
6c274a5c6ed9: Download complete 
b612cb3eb549: Download complete 
09bc2eb1d5a3: Download complete 
3d155d839dec: Download complete 
9f75ceb8178e: Download complete 
ba33e14c75b6: Download complete 
63bcc02f721d: Download complete 
2ea18e2895f5: Download complete 
e8902dd4ab2a: Download complete 
400dcdc94e4e: Download complete 
0297bfb11547: Downloading [=====>                                             ]   38.3MB/357.2MB
fbe8a1f0debc: Download complete 
138f27b0276a: Downloading 
bbb8e519ffd2: Waiting 
cec7301f8e2b: Waiting 
b757202b46b5: Waiting 
2d7d64ca32b6: Waiting 
22eb1ebbe50f: Waiting 
75c6afcffa7f: Waiting 
e50a94f94c1f: Waiting 
f2c418b39ad9: Waiting 
7ff288d55581: Waiting 
e8030d0b21df: Waiting 
unauthorized: authentication required

```

**Comment by mbufi** [comment]
@thepaulm  Did you sign in before pulling the docker image? 

1. Sign into ngc.nvidia.com using the same credentials as your NVIDIA account

2. Go to Configuration tab to generate the API key. Save this API key in a secure location. You’ll need it throughout using software.

Then to download the docker container:
1. docker login nvcr.io
2.Username: $oauthtoken
Password: [API KEY]
3. docker pull nvcr.io/nvidia/pytorch:19.04-py3

Hopefully this solves your problem.

**Comment by florischabert** [comment]
You shouldn't need to login anymore for the pytorch image. I have heard of similar issues with docker pull when the pull takes too long and times-out. Please retry the pull as a first step.

**Comment by revilokeb** [comment]
@thepaulm Authentication is not needed for that image, I have just tested it.

**Comment by florischabert** [comment]
@thepaulm Please try to pull the image again. This should work without authentication. Feel free to open a new issue if you run into problems.

**Comment by iloveaneeta45** [comment]
sorry I meet similar problems. 

When I wanna use your Retinanet, I meet following errors/problems, and I hope u might help me. There are 2 questions,

1. command dosen't work, either 
docker build -t retinanet:latest retinanet/
nvidia-docker run --rm --ipc=host -it retinanet:latest
![image](https://user-images.githubusercontent.com/40691963/60634773-c31d0580-9e42-11e9-8662-3e7d6d11e958.png)

so should I replace 'host' with my IP, or anything I did wrong? Headaches from one poor guy major in math (

2. and I seek help from , get signed up but meet errors again, by command
docker login nvcr.io
it dosen't work, a snapshot as follows
![image](https://user-images.githubusercontent.com/40691963/60634850-06777400-9e43-11e9-83cd-4e1ae4f661f0.png)

I really hope u could solve these problems ... or guide me how to use Nividia-Retinanet correctly? thx anyway if any tips (^_^)

**Comment by iloveaneeta45** [comment]
> sorry I meet similar problems.
> 
> When I wanna use your Retinanet, I meet following errors/problems, and I hope u might help me. There are 2 questions,
> 
> 1. command dosen't work, either
>    docker build -t retinanet:latest retinanet/
>    nvidia-docker run --rm --ipc=host -it retinanet:latest
>    ![image](https://user-images.githubusercontent.com/40691963/60634773-c31d0580-9e42-11e9-8662-3e7d6d11e958.png)
> 
> so should I replace 'host' with my IP, or anything I did wrong? Headaches from one poor guy major in math (
> 
> 1. and I seek help from https://ngc.nvidia.com, get signed up but meet errors again, by command
>    docker login nvcr.io
>    Username: $oauthtoken
>    Password:XXXXXXXXXXX
>    it dosen't work, a snapshot as follows
>    ![image](https://user-images.githubusercontent.com/40691963/60634850-06777400-9e43-11e9-83cd-4e1ae4f661f0.png)
> 
> I really hope u could solve these problems ... or guide me how to use Nividia-Retinanet correctly? thx anyway if any tips (^_^)

emmmmm, soory I don't notice this closed. I create a new issue proposal.

**Comment by free-soellingeraj** [comment]
I am having this problem and I can't authenticate to nvcr.io easily.  I am waiting on support to determine if it's possible for me to override the standard docker deployment command used by render.com.  
```
May 21 05:51:13 PM  ==> Cloning from https://github.com/free-soellingeraj/fastai-v3...
May 21 05:51:14 PM  ==> Checking out commit 63316d5ec0152d7ebbbb7de56ec482b7a7ba123e in branch master
May 21 05:51:17 PM  INFO[0000] Downloading base image nvcr.io/nvidia/pytorch:20.03-py3
May 21 05:51:19 PM  INFO[0002] Downloading base image nvcr.io/nvidia/pytorch:20.03-py3
May 21 05:51:21 PM  error building image: UNAUTHORIZED: authentication required; [map[Action:pull Class: Name:nvidia/pytorch Type:repository]]
May 21 05:51:21 PM  error: exit status 1
```

**Comment by lucasjinreal** [comment]
Error response from daemon: Get "https://nvcr.io/v2/": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)

**Comment by alvyC** [comment]
> Error response from daemon: Get "https://nvcr.io/v2/": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers)

@jinfagang  I am getting the same error. How did you solve the problem?
